src/server/run_sched.c

```diff
@@ -344,13 +344,19 @@ int
 schedule_high(pbs_sched *psched)
 {
 	int s;
+	attribute temp_attr;
 
 	if (psched == NULL)
 		return -1;
 
+	temp_attr.at_flags = ATR_VFLAG_SET;
+
 	if (psched->scheduler_sock == -1) {
-		if ((s = contact_sched(psched->svr_do_sched_high, NULL, psched->pbs_scheduler_addr, psched->pbs_scheduler_port)) < 0)
+		if ((s = contact_sched(psched->svr_do_sched_high, NULL, psched->pbs_scheduler_addr, psched->pbs_scheduler_port)) < 0) {
+			temp_attr.at_val.at_str = SC_DOWN;
+			sched_attr_def[(int) SCHED_ATR_sched_state].at_set(&psched->sch_attr[(int) SCHED_ATR_sched_state], &temp_attr, SET);
```

Comment:
set_str() will return an error if malloc() fails.  Do you want to catch that?  (for all other instances of setting the scheduler state as well).

Reply:
I have actually written a function set_attr_svr() which takes care of decoding, setting and clearing the attribute. I am logging appropriate error messages if corresponding functions fail.